component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPUtils.java

```diff
@@ -56,33 +62,346 @@ public static boolean isEmailOTPEnableForLocalUser(String username, Authenticati
+                apiKey = String.valueOf(context.getProperty(api + EmailOTPAuthenticatorConstants.EMAILOTP_API_KEY));
+            }
+        }
+        return apiKey;
+    }
+
+    /**
+     * Get MailingEndpoint for Gmail APIs
+     */
+    public static String getMailingEndpoint(AuthenticationContext context, Map<String, String> parametersMap,
+                                            String authenticatorName, String api) throws AuthenticationFailedException {
+        Object propertiesFromLocal = null;
+        String mailingEndpoint = null;
+        String tenantDomain = context.getTenantDomain();
+        if (!tenantDomain.equals(EmailOTPAuthenticatorConstants.SUPER_TENANT)) {
+            IdentityHelperUtil.loadApplicationAuthenticationXMLFromRegistry(context, authenticatorName, tenantDomain);
+            propertiesFromLocal = context.getProperty(IdentityHelperConstants.GET_PROPERTY_FROM_REGISTRY);
+        }
+        if (propertiesFromLocal != null || tenantDomain.equals(EmailOTPAuthenticatorConstants.SUPER_TENANT)) {
+            if (parametersMap.containsKey(api + EmailOTPAuthenticatorConstants.MAILING_ENDPOINT)) {
+                mailingEndpoint = parametersMap.get(api + EmailOTPAuthenticatorConstants.MAILING_ENDPOINT);
+            }
+        } else {
+            if ((context.getProperty(api + EmailOTPAuthenticatorConstants.MAILING_ENDPOINT)) != null) {
+                mailingEndpoint = String.valueOf(context.getProperty
+                        (api + EmailOTPAuthenticatorConstants.MAILING_ENDPOINT));
+            }
+        }
+        return mailingEndpoint;
+    }
+
+    /**
+     * Get required payload for Gmail APIs
+     */
+    public static String getPreparePayload(AuthenticationContext context, Map<String, String> parametersMap,
+                                           String authenticatorName, String api) throws AuthenticationFailedException {
+        Object propertiesFromLocal = null;
+        String payload = null;
+        String tenantDomain = context.getTenantDomain();
+        if (!tenantDomain.equals(EmailOTPAuthenticatorConstants.SUPER_TENANT)) {
+            IdentityHelperUtil.loadApplicationAuthenticationXMLFromRegistry(context, authenticatorName, tenantDomain);
+            propertiesFromLocal = context.getProperty(IdentityHelperConstants.GET_PROPERTY_FROM_REGISTRY);
+        }
+        if (propertiesFromLocal != null || tenantDomain.equals(EmailOTPAuthenticatorConstants.SUPER_TENANT)) {
+            if (parametersMap.containsKey(api + EmailOTPAuthenticatorConstants.PAYLOAD)) {
+                payload = parametersMap.get(api + EmailOTPAuthenticatorConstants.PAYLOAD);
+            }
+        } else {
+            if ((context.getProperty(api + EmailOTPAuthenticatorConstants.PAYLOAD)) != null) {
+                payload = String.valueOf(context.getProperty(api + EmailOTPAuthenticatorConstants.PAYLOAD));
```

Comment:
When we get the parameter from context, it returns as object. So that, cast the parameter to String.